src/test/java/net/uaznia/lukanus/hudson/plugins/gitparameter/UiAcceptanceTest.java

```diff
@@ -0,0 +1,95 @@
+package net.uaznia.lukanus.hudson.plugins.gitparameter;
+
+import hudson.model.FreeStyleProject;
+import io.github.bonigarcia.wdm.WebDriverManager;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.tools.ant.filters.StringInputStream;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.support.ui.Select;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import javax.xml.transform.Source;
+import javax.xml.transform.stream.StreamSource;
+import java.io.File;
+import java.nio.file.Files;
+import java.time.Duration;
+
+import static org.junit.Assert.assertTrue;
+
+public class UiAcceptanceTest {
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    private WebDriver driver;
+
+    @BeforeClass
+    public static void setUpClass() {
+        if (isCi()) {
+            // The browserVersion needs to match what is provided by the Jenkins Infrastructure
+            // If you see an exception like this:
+            //
+            // org.openqa.selenium.SessionNotCreatedException: Could not start a new session. Response code 500. Message: session not created: This version of ChromeDriver only supports Chrome version 114
+            // Current browser version is 112.0.5615.49 with binary path /usr/bin/chromium-browser
+            //
+            // Then that means you need to update the version here to match the current browser version.
+            WebDriverManager.chromedriver().browserVersion("112").setup();
+        } else {
+            WebDriverManager.chromedriver().setup();
+        }
+    }
+
+    private static boolean isCi() {
+        return StringUtils.isNotBlank(System.getenv("CI"));
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        FreeStyleProject freeStyleProject = j.createFreeStyleProject("test");
+        String config = Files.readString(new File("src/test/resources/test-config.xml").toPath());;
+        config = config.replace("${GIT_URL}", new File(".").getAbsolutePath());
+
+        freeStyleProject.updateByXml((Source) new StreamSource(new StringInputStream(config)));
```

Comment:
The usual way this is done in the Jenkins test harness is to put the XML file in e.g. `src/test/resources/net/uaznia/lukanus/hudson/plugins/gitparameter/UiAcceptanceTest/test/jobs/test/config.xml` and add `@LocalData` above `@Test`. The test harness with load Jenkins with the given XML configuration. See the Javadoc for `@LocalData` for more information and see the `config.xml` files in the Jenkins core test suite for examples.